File: src/python/services/phi.py
import requests

class Phi:

    def __init__(
            self,
            deployment_endpoint: str,
            deployment_key: str
    ):
        self._deployment_endpoint = deployment_endpoint
        self._deployment_key = deployment_key

    def create_completion(
            self,
            messages: list[dict]
    ):
        response = requests.post(
            f"{self._deployment_endpoint}",
            headers={
                "Authorization": f"Bearer {self._deployment_key}"
            },
            json={
                "messages": messages
            }
        )

        response.raise_for_status()

        return response.json()

File: src/node/services/phi.js
class Phi {
    constructor(deploymentEndpoint, deploymentKey) {
        this.deploymentEndpoint = deploymentEndpoint;
        this.deploymentKey = deploymentKey;
    }

    async createCompletion(messages) {
        try {
            const response = await fetch(
                `${this.deploymentEndpoint}`,
                {
                    method: 'POST',
                    headers: {
                        'Authorization': `Bearer ${this.deploymentKey}`,
                        'Content-Type': 'application/json'
                    },
                    body: JSON.stringify({ messages: messages })
                }
            );

            if (!response.ok) {
                throw new Error(`HTTP error! status: ${response.status}`);
            }

            return await response.json();
        } catch (error) {
            console.error(`Error in createCompletion: ${error.message}`);
        }
    }
}

module.exports.Phi = Phi;